### PR TITLE
Support for zcli themes:update

### PR DESCRIPTION
**Pull request by luis-almeida**
Adding a new `zcli themes:update` command to update a previously imported theme.

### [`1146b45`](https://github.com/zendesk/zcli/pull/182/commits/1146b45fa8ea41e10a2fd6877bfd63d4ee59ce50) Apply suggestions from code review

Co-authored-by: André Alves <aalves@zendesk.com>

### [`19aeab5`](https://github.com/zendesk/zcli/pull/182/commits/19aeab53b8630ef839a5566e8edea718d3475243) Update docs/themes.md

Co-authored-by: André Alves <aalves@zendesk.com>

## Checklist

- [x] :guardsman: includes new unit and functional tests

### CODE REVIEW

File: docs/themes.md [hunk 1]
```diff
@@ -56,3 +58,23 @@ EXAMPLES
...
+
+ARGUMENTS
+  THEMEDIRECTORY  [default: .] theme path where manifest.json exists
+
+OPTIONS
+  --themeId               The id of the theme to update
+  --replaceSettings       Whether or not to replace the current theme settings
+
+EXAMPLES
+  $ zcli themes:import ./copenhagen_theme --themeId=123456789100
```
Comment (anpa) [inline docs/themes.md:78]:
```suggestion
  $ zcli themes:update ./copenhagen_theme --themeId=123456789100
```

File: docs/themes.md [hunk 2]
```diff
@@ -56,3 +58,23 @@ EXAMPLES
...
+  $ zcli themes:import ./copenhagen_theme --themeId=123456789100 --replaceSettings
```
Comment (anpa) [inline docs/themes.md:79]:
```suggestion
  $ zcli themes:update ./copenhagen_theme --themeId=123456789100 --replaceSettings
```

File: packages/zcli-themes/src/commands/themes/update.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    themeId: Flags.string({ description: 'The id of the theme to update' }),
+    replaceSettings: Flags.boolean({ default: false, description: 'Whether or not to replace the current theme settings' })
+  }
+
+  static args = [
+    { name: 'themeDirectory', required: true, default: '.' }
+  ]
+
+  static examples = [
+    '$ zcli themes:import ./copenhagen_theme'
```
Comment (anpa) [inline packages/zcli-themes/src/commands/themes/update.ts:22]:
```suggestion
    '$ zcli themes:update ./copenhagen_theme' --themeId=123456789100
```

File: packages/zcli-themes/src/commands/themes/update.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  ]
```
Comment (anpa) [inline packages/zcli-themes/src/commands/themes/update.ts:23]: nit: Should we also add the example with the `--replaceSettings` flag?

File: packages/zcli-themes/src/commands/themes/update.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  static strict = false
+
+  async run () {
+    let { flags: { themeId, replaceSettings }, argv: [themeDirectory] } = await this.parse(Update)
+    const themePath = path.resolve(themeDirectory)
+
+    themeId = themeId || await CliUx.ux.prompt('Theme ID')
```
Comment (anpa) [inline packages/zcli-themes/src/commands/themes/update.ts:31]: nit: for the `import` command we are using `inquirer` to [prompt the brand Id](https://github.com/zendesk/zcli/blob/master/packages/zcli-themes/src/lib/getBrandId.ts#L16). Should they be consistent?

File: packages/zcli-themes/src/commands/themes/update.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    try {
+      await uploadThemePackage(job, readStream)
+    } finally {
+      removePackage()
+    }
+
+    await pollJobStatus(themePath, job.id)
+
+    this.log(chalk.green('Theme updated successfully'))
```
Comment (anpa) [inline packages/zcli-themes/src/commands/themes/update.ts:44]:
```suggestion
 this.log(chalk.green('Theme imported successfully'), `theme ID: ${job.data.theme_id}`)
```

For consistency with [import](https://github.com/zendesk/zcli/blob/master/packages/zcli-themes/src/commands/themes/import.ts#L45) and [publish](https://github.com/zendesk/zcli/pull/181/files#diff-c5c2ff46ec004beb410f94a963da84cf7eb3a729d40e780fc56020daa82398eeR37)?

File: packages/zcli-themes/src/lib/createThemeImportJob.ts [hunk 7]
```diff
@@ -28,11 +27,7 @@ export default async function createThemeImportJob (brandId: string): Promise<Pe
...
   } catch (e) {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const errors = (e as any).response.data.errors
-    for (const { code, title } of errors) {
-      error(`${chalk.bold(code)} - ${title}`)
-    }
-
-    error(new CLIError(e as axios.AxiosError))
+    const [{ code, title }] = (e as any).response.data.errors
+    error(`${chalk.bold(code)} - ${title}`)
```
Comment (anpa) [inline packages/zcli-themes/src/lib/createThemeImportJob.ts:31]: Curious, why this change?

**Review by anpa — COMMENTED** [review]
Left some comments but it's looking goooood! :excite_x2:

File: packages/zcli-themes/src/commands/themes/update.ts (hunk 5 above)
Comment (luis-almeida) [reply packages/zcli-themes/src/commands/themes/update.ts:31]: Updating does not require a `brandId` as the theme is already imported in a brand so we already have a record.
So we just need the `themeId` which we also prompt for. We don't need the `inquirer` here because there's no options to select from so `CliUx.ux.prompt` is enough.

File: packages/zcli-themes/src/lib/createThemeImportJob.ts (hunk 7 above)
Comment (luis-almeida) [reply packages/zcli-themes/src/lib/createThemeImportJob.ts:31]: Once we call `error` it will error out, and to avoid that we need extra logic.
Even though we get an array of errors, it seems to only ever include one, so to simplify the code I'm just picking the first one.

File: packages/zcli-themes/src/commands/themes/update.ts (hunk 5 above)
Comment (anpa) [reply packages/zcli-themes/src/commands/themes/update.ts:32]: Ahhh, gotcha! Thanks for explaining!

File: docs/themes.md [hunk 8]
```diff
@@ -56,3 +58,23 @@ EXAMPLES
...
+```
+USAGE
+  $ zcli themes:update [THEMEDIRECTORY]
+
+ARGUMENTS
+  THEMEDIRECTORY  [default: .] theme path where manifest.json exists
+
+OPTIONS
+  --themeId               The id of the theme to update
+  --replaceSettings       Whether or not to replace the current theme settings
```
Comment (anpa) [inline docs/themes.md:75]: nit: sorry, I missed this one in the first round of review 🙈 
 
```suggestion
  --replaceSettings       [default: false] Whether or not to replace the current theme settings
```

File: packages/zcli-themes/src/commands/themes/update.ts (hunk 6 above)
Comment (anpa) [reply packages/zcli-themes/src/commands/themes/update.ts:45]: Sorry I forgot to mark my comment as a nit pick 🙈  It was more because we also output the theme ID for [publish](https://github.com/zendesk/zcli/pull/181/files#diff-c5c2ff46ec004beb410f94a963da84cf7eb3a729d40e780fc56020daa82398eeR37) even though the id is an argument. It's a nit pick though.

**Review by anpa — APPROVED** [review]
🚀 🚢 :shipit: 💥 🥳
